qiskit/quantum_info/analysis/distance.py

```diff
@@ -14,11 +14,50 @@
 import numpy as np
 
 
+def hellinger_distance(dist_p, dist_q):
+    """Computes the Hellinger distance between
+    two counts distributions.
+
+    Parameters:
+        dist_p (dict): First dict of counts.
+        dist_q (dict): Second dict of counts.
+
+    Returns:
+        float: Distance
+
+    References:
+        `Hellinger Distance @ wikipedia <https://en.wikipedia.org/wiki/Hellinger_distance>`_
+    """
+    p_sum = sum(dist_p.values())
+    q_sum = sum(dist_q.values())
+
+    p_normed = {}
+    for key, val in dist_p.items():
+        p_normed[key] = val/p_sum
+
+    q_normed = {}
+    for key, val in dist_q.items():
+        q_normed[key] = val/q_sum
+
+    total = 0
+    for key, val in p_normed.items():
+        if key in q_normed.keys():
+            total += (np.sqrt(val) - np.sqrt(q_normed[key]))**2
+            del q_normed[key]
+        else:
+            total += val
+    total += sum(q_normed.values())
+
+    dist = np.sqrt(total)/np.sqrt(2)
+
+    return dist
+
+
 def hellinger_fidelity(dist_p, dist_q):
     """Computes the Hellinger fidelity between
     two counts distributions.
 
-    The fidelity is defined as 1-H where H is the
+    The fidelity is defined as (1-H**2)**2 where H is the
```

Comment:
It would be good to explicitly state in the documentation that  (the new definition of) the Helliger fidelity of two probability distributions P, Q is defined so that it is equal to the quantum state fidelity F_H(P, Q) = F(\rho_P, \rho_Q) of the diagonal density matrices :math:`\rho_P = \sum_i p_i |i \rangle\langle i|`, :math:`\rho_Q = \sum_i q_i |i \rangle\langle i|`